File: generadorContrasenias/GeneradorContrasenias.java
package generadorContrasenias;

import java.util.Random;
import java.util.Scanner;

public class GeneradorContrasenias {
    public static String generar(int longitud, boolean usarMayus, boolean usarNum, boolean usarSimb) {
        String minusculas = "abcdefghijklmnopqrstuvwxyz";
        String mayusculas = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        String numeros = "0123456789";
        String simbolos = "!@#$%^&*()-_=+<>?";

        String caracteres = minusculas;
        if (usarMayus) caracteres += mayusculas;
        if (usarNum) caracteres += numeros;
        if (usarSimb) caracteres += simbolos;

        StringBuilder contrasena = new StringBuilder();
        Random rand = new Random();

        for (int i = 0; i < longitud; i++) {
            contrasena.append(caracteres.charAt(rand.nextInt(caracteres.length())));
        }
        return contrasena.toString();
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int longitud;
        char mayus, num, simb;

        System.out.print("Longitud (8-16): ");
        longitud = sc.nextInt();
        System.out.print("Incluir mayusculas? (s/n): ");
        mayus = sc.next().charAt(0);
        System.out.print("Incluir numeros? (s/n): ");
        num = sc.next().charAt(0);
        System.out.print("Incluir simbolos? (s/n): ");
        simb = sc.next().charAt(0);

        System.out.println("Contraseña generada: " +
            generar(longitud, mayus=='s', num=='s', simb=='s'));
    }
}

File: generadorContrasenias/GeneradorContrasenias.cpp
#include <iostream>
#include <string>
#include <cstdlib>
#include <ctime>

using namespace std;

string generarContrasena(int longitud, bool usarMayus, bool usarNum, bool usarSimbolos) {
    string minusculas = "abcdefghijklmnopqrstuvwxyz";
    string mayusculas = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    string numeros = "0123456789";
    string simbolos = "!@#$%^&*()-_=+<>?";

    string caracteres = minusculas;
    if (usarMayus) caracteres += mayusculas;
    if (usarNum) caracteres += numeros;
    if (usarSimbolos) caracteres += simbolos;

    string contrasena;
    for (int i = 0; i < longitud; i++) {
        contrasena += caracteres[rand() % caracteres.size()];
    }
    return contrasena;
}

int main() {
    srand(time(0));
    int longitud;
    char mayus, num, simb;

    cout << "Longitud (8-16): ";
    cin >> longitud;
    cout << "Incluir mayusculas? (s/n): ";
    cin >> mayus;
    cout << "Incluir numeros? (s/n): ";
    cin >> num;
    cout << "Incluir simbolos? (s/n): ";
    cin >> simb;

    cout << "Contrasenia generada: " 
         << generarContrasena(longitud, mayus=='s', num=='s', simb=='s') << endl;
    return 0;
}
